rust/agama-server/src/web/common.rs

```diff
@@ -165,10 +165,21 @@ struct ProgressState<'a> {
     proxy: ProgressProxy<'a>,
 }
 
-async fn progress(State(state): State<ProgressState<'_>>) -> Result<Json<Progress>, Error> {
+/// Information about the current progress sequence.
+#[derive(Clone, Default, Serialize)]
+pub struct ProgressSequence {
```

Comment:
Why not to add a `steps` field to the `Progress` struct and fill it in `#from_proxy`?

Reply:
Because `Progress` represents a progress change, it is used in the "Progress" signal, and it does not include the list of steps.